.github/workflows/release.yml

```diff
@@ -5,141 +5,246 @@ on:
-          PRE_RELEASE: "false"
-          CHANGELOG_FILE: "CHANGELOG.md"
-          ALLOW_EMPTY_CHANGELOG: "false"
+          mv crab-runtime/*.compact.wasm ../deploy/
+          mv crab-runtime/*.compact.json ../deploy/
+          mv darwinia-runtime/*.compact.wasm ../deploy/
+          mv darwinia-runtime/*.compact.json ../deploy/
+
+## not open for now. https://github.com/darwinia-network/darwinia/pull/779#issuecomment-964911852
+#          mv crab-runtime/*.compact.compressed.wasm ../deploy/
+#          mv crab-runtime/*.compact.compressed.json ../deploy/
+#          mv darwinia-runtime/*.compact.compressed.wasm ../deploy/
+#          mv darwinia-runtime/*.compact.compressed.json ../deploy/
+
+      - name: Hash file
+        run: |
+          cd deploy
+          md5sum * > ../md5sums.txt
+          mv ../md5sums.txt ./
+
+      - uses: olegtarasov/get-tag@v2.1
+        id: tag-name
+      - uses: benjlevesque/short-sha@v1.2
+        id: short-sha
         with:
-          args: |
-            deploy/bin/*
+          length: 7
 
-      - uses: 8398a7/action-slack@v3
-        if: failure()
+      - name: Generate template data
+        run: |
+          PROPOSAL_CRAB_COMPACT=$(cat deploy/crab-runtime.compact.json | jq -r .proposal_hash)
+          PROPOSAL_DARWINIA_COMPACT=$(cat deploy/darwinia-runtime.compact.json | jq -r .proposal_hash)
+
+          # # not open for now. https://github.com/darwinia-network/darwinia/pull/779#issuecomment-964911852
+          # PROPOSAL_CRAB_COMPRESSED=$(cat deploy/crab-runtime.compact.compressed.json | jq -r .proposal_hash)
+          # PROPOSAL_DARWINIA_COMPRESSED=$(cat deploy/darwinia-runtime.compressed.compact.json | jq -r .proposal_hash)
+
+          IMAGE_NAME=darwinia
+          IMAGE_PREFIX=quay.io/darwinia-network
+          IMAGE_PUSH_NAME_TAG=${IMAGE_PREFIX}/${IMAGE_NAME}:${GIT_TAG_NAME}
+          IMAGE_PUSH_NAME_SHA=${IMAGE_PREFIX}/${IMAGE_NAME}:sha-${SHA}
+
+          JSON=$( jq -n \
+            --arg crab_proposal_compact "$PROPOSAL_CRAB_COMPACT" \
+            --arg darwinia_proposal_compact "$PROPOSAL_DARWINIA_COMPACT" \
+            --arg image_tag "$IMAGE_PUSH_NAME_TAG" \
+            --arg image_sha "$IMAGE_PUSH_NAME_SHA" \
+            '{
+              crab_proposal_compact: $crab_proposal_compact,
+              darwinia_proposal_compact: $darwinia_proposal_compact,
+              image_tag: $image_tag,
+              image_sha: $image_sha
+            }' )
+          echo "$JSON" > shared/template_data.json
+
+      - name: Render release page
+        uses: jayamanikharyono/jinja-action@v0.1
         with:
-          status: ${{ job.status }}
-          fields: repo,author,eventName,workflow,ref,commit
-        env:
-          SLACK_WEBHOOK_URL: ${{ secrets.SLACK_INCOMING_WEBHOOK_URL }}
+          datafile: shared/template_data.json
+          path: .maintain/release-template.md
+
+      - name: Publish github release
+        uses: ncipollo/release-action@v1
```

Comment:
Any reason not using the more common one? https://github.com/softprops/action-gh-release